python/ray/serve/tests/test_fastapi.py

```diff
@@ -490,6 +490,35 @@ class TestDeployment:
     assert resp.json() == {"a": "a", "b": ["b"]}
 
 
+def test_fastapiwrapper_constructor_before_startup_hooks(serve_instance):
+    """
+    Tests that the class constructor is called before the startup hooks
+    are run in `FastAPIWrapper`.
+    """
+    app = FastAPI()
+
+    @serve.deployment(route_prefix="/")
+    @serve.ingress(app)
+    class TestDeployment:
+        def __init__(self):
+            try:
+                # `self.lifespan` is defined in `FastAPIWrapper` before the
```

Comment:
🤦 it occurs to me that if user define `self.lifespan` and `self.app` our fastapi integration will cease to work. Can we rename the `self.app` and self.lifespan` to `self._serve_app` and `self._serve_asgi_lifespan` to prevent confusion? 

This will increase the scope of this PR a bit but it should bestraightforwardd renaming changes.

Reply:
Actually, with the new test should we keep `self.lifespan` and `self.app`? If the user defines either of these, they will be overwritten in the `FastAPIWrapper.__init__` method